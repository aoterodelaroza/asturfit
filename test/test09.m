#! /usr/bin/octave -q
% Copyright (C) 2010 Victor Lua~na and Alberto Otero-de-la-Roza
%
% This octave routine is free software: you can redistribute it and/or modify
% it under the terms of the GNU General Public License as published by
% the Free Software Foundation, either version 3 of the License, or (at
% your option) any later version. See <http://www.gnu.org/licenses/>.
%
% The routine distributed in the hope that it will be useful, but WITHOUT
% ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
% FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
% more details.

% Test09: Detecting jumps in a dataset

addpath("../src/");

# read in a smooth data set
[v,e] = readEVdata('mgo-sety1.dat',1);
vref = median(v);

# now we add artificially some jumps
# (keep the original data for further comparison)
vmin = min(v); vmax = max(v); vrange = vmax-vmin;
vj = v; ej = e;
jumps.v = [vmin+vrange/3, vmin+2*vrange/3];
jumps.e = [0.010, -0.011];
for i = 1 : length(jumps.v)
   printf('Adding a jump of %.6f for V>%.6f\n', jumps.e(i), jumps.v(i));
   rk = find(vj > jumps.v(i));
   ej(rk) += jumps.e(i);
endfor

# let's do the automatic detection
[st,ecorr] = guessjumps3(vj,ej,:,1);
fileplot = 'test09-ev.eps';
print(fileplot, '-FHelvetica:24', '-depsc');
printf('See the E(V) curve in the file: %s\n', fileplot);

# compare the original and the corrected energies
SSerr = sum((e-ecorr).^2);
SStot = sum((e-mean(e)).^2);
R2 = 1 - SSerr / SStot;
printf('\n');
printf('Global error of the corrected data:\n');
printf('Determination coef. (R2, 1 for a exact correction): %.9e\n', R2);
printf('1-R2: %.2e\n', SSerr / SStot);
