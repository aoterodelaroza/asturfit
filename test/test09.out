warning: mark_as_command is obsolete and will be removed from a future version of Octave
readEVdata: Reading input file -> mgo-sety1.dat
Input volume unit: bohr3
Input energy unit: ry
* Energy and volume ranges 
  V-range (bohr^3) : [71.5195563798 , 143.0391127595] dV = 71.5195563796
  E-range (hartree) : [-17.1721409750 , -17.0038769900] dE = 0.1682639850
  Number of data points: 129
  Number of electrons: 20

Adding a jump of 0.010000 for V>95.359409
Adding a jump of -0.011000 for V>119.199261
DBG(gj3):
Interpolation error (mean,stddev): 2.173e-05 9.296e-04
Jumps detected: 2
       -jump- ----Vjump--- -----Ejump----
     1      1    95.266284   1.000103e-02
     2      2   119.292385  -1.100077e-02

Plot file produced: gj3-step3.eps
See the E(V) curve in the file: test09-ev.eps

Global error of the corrected data:
Determination coef. (R2, 1 for a exact correction): 9.999999998e-01
1-R2: 1.87e-10
